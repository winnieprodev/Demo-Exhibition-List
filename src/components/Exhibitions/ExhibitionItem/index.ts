export { default as ExhibitionItem } from "./ExhibtionItem";
