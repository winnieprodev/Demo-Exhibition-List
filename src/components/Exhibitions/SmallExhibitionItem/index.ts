export { default as SmallExhibitionItem } from "./SmallExhibitionItem";
