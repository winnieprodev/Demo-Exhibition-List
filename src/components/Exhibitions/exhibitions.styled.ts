import styled from "styled-components";
import {
  secondaryColorLighter,
  secondaryColorDarker,
} from "styles/colors";

export const Wrapper = styled.div`
  width: 100%;
  display: flex;
  flex-wrap: wrap;
  justify-content: center;
`;

export const SmallWrapper = styled.div`
  width: 100%;
  display: none;
  flex-wrap: wrap;
  justify-content: center;
  flex-direction: column;
  @media (max-width: 716px) {
    display: flex;
  }
`;

export const Table = styled.table`
  border-collapse: collapse;
  td,
  th {
    border: 2px solid ${secondaryColorDarker};
    padding: 10px;
    text-align: center;
  }

  thead > tr {
    background-color: ${secondaryColorLighter};
    cursor: pointer;
  }
  
  display: block;
  @media (max-width: 716px) {
    display: none;
  }
`;

export const Icon = styled.div`
  padding-left: 12px;
  color: rgba(0, 0, 0, .4);
`;

export const TableSortLabel = styled.div`
  display: flex;
  align-items: center;
  justify-content: center;
`;