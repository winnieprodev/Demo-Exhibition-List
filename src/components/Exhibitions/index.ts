export { default as Exhibitions } from "./Exhibitions";