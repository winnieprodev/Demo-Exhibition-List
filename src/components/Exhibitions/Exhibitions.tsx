import React from "react";
import * as Styled from "./exhibitions.styled";
import { ExhibitionItem } from "./ExhibitionItem";
import { ExhibitionData } from "models";
import { SmallExhibitionItem } from "./SmallExhibitionItem";
import { library as iconLibrary } from "@fortawesome/fontawesome-svg-core";
import {
    faSortDown,
    faSortUp
} from "@fortawesome/free-solid-svg-icons";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import { SortingDirection, SortingObject, SortingSource } from "pages/ExhibitionList/ExhibitionList";

iconLibrary.add(faSortDown);
iconLibrary.add(faSortUp);

interface OwnProps {
  exhibitions: ExhibitionData[];
  sortingColumns: SortingObject[];
  handleSort: (sortObject: string) => void;
}

const Exhibitions = ({ handleSort, exhibitions, sortingColumns }: React.PropsWithChildren<OwnProps>) => {
  const renderSmallContent = () => {
    return (
      <Styled.SmallWrapper>
        {
          exhibitions.map(item => (
            <Exhibitions.SmallItem data={item} key={item.id}/>
          ))
        }
      </Styled.SmallWrapper>
    )
  }

  const titleSorting = sortingColumns.find(x => x.source === SortingSource.Title);
  const descriptionSorting = sortingColumns.find(x => x.source === SortingSource.Description);
  const isFeaturedSorting = sortingColumns.find(x => x.source === SortingSource.IsFeatured);
  const galleryTitleSorting = sortingColumns.find(x => x.source === SortingSource.GalleryTitle);
  const exhibitionTypeSorting = sortingColumns.find(x => x.source === SortingSource.Type);

  return (
    <React.Fragment>
      <Styled.Wrapper>
        <Styled.Table>
          <thead>
            <tr>
              <th className="text-center" style={{ width: '20%' }} onClick={() => handleSort("title")}>
                <Styled.TableSortLabel>
                  Title
                  <Styled.Icon>
                    {titleSorting?.direction === SortingDirection.Asc ? <FontAwesomeIcon icon="sort-down" /> : <FontAwesomeIcon icon="sort-up" />}
                  </Styled.Icon>
                </Styled.TableSortLabel>
              </th>
              <th className="text-center" style={{ width: '30%' }} onClick={() => handleSort("description")}>
                <Styled.TableSortLabel>
                  Description
                  <Styled.Icon>
                    {descriptionSorting?.direction === SortingDirection.Asc ? <FontAwesomeIcon icon="sort-down" /> : <FontAwesomeIcon icon="sort-up" />}
                  </Styled.Icon>
                </Styled.TableSortLabel>
              </th>
              <th className="text-center" style={{ width: '10%' }} onClick={() => handleSort("is_featured")}>
                <Styled.TableSortLabel>
                  Is Featured
                  <Styled.Icon>
                    {isFeaturedSorting?.direction === SortingDirection.Asc ? <FontAwesomeIcon icon="sort-down" /> : <FontAwesomeIcon icon="sort-up" />}
                  </Styled.Icon>
                </Styled.TableSortLabel>
              </th>
              <th className="text-center" style={{ width: '20%' }} onClick={() => handleSort("gallery_title")}>
                <Styled.TableSortLabel>
                  Gallery Title
                  <Styled.Icon>
                    {galleryTitleSorting?.direction === SortingDirection.Asc ? <FontAwesomeIcon icon="sort-down" /> : <FontAwesomeIcon icon="sort-up" />}
                  </Styled.Icon>
                </Styled.TableSortLabel>
              </th>
              <th className="text-center" style={{ width: '20%' }} onClick={() => handleSort("type")}>
                <Styled.TableSortLabel>
                  Type of Exhibition
                  <Styled.Icon>
                    {exhibitionTypeSorting?.direction === SortingDirection.Asc ? <FontAwesomeIcon icon="sort-down" /> : <FontAwesomeIcon icon="sort-up" />}
                  </Styled.Icon>
                </Styled.TableSortLabel>
              </th>
            </tr>
          </thead>
          <tbody>
            {
              exhibitions.map(item => (
                <Exhibitions.ExhibitionItem info={item} key={item.id}/>
              ))
            }
          </tbody>
        </Styled.Table>
        {renderSmallContent()}
      </Styled.Wrapper>
    </React.Fragment>
  )
};

Exhibitions.ExhibitionItem = ExhibitionItem;
Exhibitions.SmallItem = SmallExhibitionItem;

export default Exhibitions;
