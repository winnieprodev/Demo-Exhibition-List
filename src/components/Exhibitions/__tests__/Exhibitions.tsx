import { render } from "@testing-library/react";

import Exhibitions from "../Exhibitions";

const mockExhibitions = [
  {
    "id": 3070,
    "api_model": "exhibitions",
    "api_link": "https://api.artic.edu/api/v1/exhibitions/3070",
    "title": "Van Gogh",
    "is_featured": false,
    "short_description": "Please note: this exhibition is open on weekdays only.",
    "web_url": "https://nocache.www.artic.edu/exhibitions/3070/van-gogh",
    "image_url": null,
    "status": "Closed",
    "aic_start_at": "2016-02-16T00:00:00-06:00",
    "aic_end_at": "2016-05-09T00:00:00-05:00",
    "gallery_id": 2147472240,
    "gallery_title": "Ryerson Library Reading Room",
    "artwork_ids": [],
    "artwork_titles": [],
    "artist_ids": [],
    "site_ids": [],
    "image_id": null,
    "alt_image_ids": [],
    "document_ids": [],
    "suggest_autocomplete_all": {
        "input": [
            "Van Gogh"
        ],
        "contexts": {
            "groupings": [
                "title"
            ]
        }
    },
    "source_updated_at": "2019-05-10T15:52:37-05:00",
    "updated_at": "2022-09-10T23:41:40-05:00",
    "timestamp": "2022-09-11T09:02:57-05:00"
  },
  {
    "id": 3071,
    "api_model": "exhibitions",
    "api_link": "https://api.artic.edu/api/v1/exhibitions/3071",
    "title": "Van Gogh1",
    "is_featured": false,
    "short_description": "Please note: this exhibition is open on weekdays only.",
    "web_url": "https://nocache.www.artic.edu/exhibitions/3070/van-gogh",
    "image_url": null,
    "status": "Closed",
    "aic_start_at": "2016-02-16T00:00:00-06:00",
    "aic_end_at": "2016-05-09T00:00:00-05:00",
    "gallery_id": 2147472240,
    "gallery_title": "Ryerson Library Reading Room",
    "artwork_ids": [],
    "artwork_titles": [],
    "artist_ids": [],
    "site_ids": [],
    "image_id": null,
    "alt_image_ids": [],
    "document_ids": [],
    "suggest_autocomplete_all": {
        "input": [
            "Van Gogh"
        ],
        "contexts": {
            "groupings": [
                "title"
            ]
        }
    },
    "source_updated_at": "2019-05-10T15:52:37-05:00",
    "updated_at": "2022-09-10T23:41:40-05:00",
    "timestamp": "2022-09-11T09:02:57-05:00"
  }
]

describe("Exhibitions", () => {
  describe("Snapshot tests", () => {
    test("it matches the snapshot for basic use", () => {
      const { container } = render(
        <Exhibitions handleSort={jest.fn} exhibitions={mockExhibitions} />
      );
      expect(container.firstChild).toMatchSnapshot();
    });
  });
});
