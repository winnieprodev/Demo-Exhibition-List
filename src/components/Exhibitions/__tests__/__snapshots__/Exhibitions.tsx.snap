// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Exhibitions Snapshot tests it matches the snapshot for basic use 1`] = `
.c0 {
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
}

.c4 {
  width: 100%;
  display: none;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c1 {
  border-collapse: collapse;
  display: block;
}

.c1 td,
.c1 th {
  border: 2px solid #E7E7E7;
  padding: 10px;
  text-align: center;
}

.c1 thead > tr {
  background-color: #F3F3F3;
  cursor: pointer;
}

.c2 {
  background-color: #f4d5d5;
  min-height: 60px;
  height: 60px;
}

.c3 {
  overflow: hidden;
  text-overflow: ellipsis;
  -webkit-line-clamp: 2;
  display: -webkit-box;
  -webkit-box-orient: vertical;
}

.c5 {
  background-color: #f4d5d5;
  padding: 16px;
  margin-bottom: 16px;
  border-radius: 10px;
}

.c6 {
  padding: 4px;
  width: 100%;
}

.c7 {
  font-size: 16px;
  font-weight: 600;
  margin-right: 8px;
}

.c8 {
  padding-left: 16px;
  overflow: hidden;
  text-overflow: ellipsis;
  -webkit-line-clamp: 2;
  display: -webkit-box;
  -webkit-box-orient: vertical;
}

@media (max-width:716px) {
  .c4 {
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
  }
}

@media (max-width:716px) {
  .c1 {
    display: none;
  }
}

<div
  class="c0"
>
  <table
    class="c1"
  >
    <thead>
      <tr>
        <th
          class="text-center"
          style="width: 20%;"
        >
          Title
        </th>
        <th
          class="text-center"
          style="width: 30%;"
        >
          Description
        </th>
        <th
          class="text-center"
          style="width: 10%;"
        >
          Is Featured
        </th>
        <th
          class="text-center"
          style="width: 20%;"
        >
          Gallery Title
        </th>
        <th
          class="text-center"
          style="width: 20%;"
        >
          Type of Exhibition
        </th>
      </tr>
    </thead>
    <tbody>
      <tr
        class="c2"
      >
        <td>
          <div
            class="c3"
          >
            Van Gogh
          </div>
        </td>
        <td>
          <div
            class="c3"
          />
        </td>
        <td>
          No
        </td>
        <td>
          Ryerson Library Reading Room
        </td>
        <td />
      </tr>
      <tr
        class="c2"
      >
        <td>
          <div
            class="c3"
          >
            Van Gogh1
          </div>
        </td>
        <td>
          <div
            class="c3"
          />
        </td>
        <td>
          No
        </td>
        <td>
          Ryerson Library Reading Room
        </td>
        <td />
      </tr>
    </tbody>
  </table>
  <div
    class="c4"
  >
    <div
      class="c5"
    >
      <div
        class="c6"
      >
        <div
          class="c7"
        >
          Title
        </div>
        <div
          class="c8"
        >
          Van Gogh
        </div>
      </div>
      <div
        class="c6"
      >
        <div
          class="c7"
        >
          Description
        </div>
        <div
          class="c8"
        />
      </div>
      <div
        class="c6"
      >
        <div
          class="c7"
        >
          Is Featured
        </div>
        <div
          class="c8"
        >
          No
        </div>
      </div>
      <div
        class="c6"
      >
        <div
          class="c7"
        >
          Gallery Title
        </div>
        <div
          class="c8"
        >
          Ryerson Library Reading Room
        </div>
      </div>
      <div
        class="c6"
      >
        <div
          class="c7"
        >
          Type of Exhibition
        </div>
        <div
          class="c8"
        />
      </div>
    </div>
    <div
      class="c5"
    >
      <div
        class="c6"
      >
        <div
          class="c7"
        >
          Title
        </div>
        <div
          class="c8"
        >
          Van Gogh1
        </div>
      </div>
      <div
        class="c6"
      >
        <div
          class="c7"
        >
          Description
        </div>
        <div
          class="c8"
        />
      </div>
      <div
        class="c6"
      >
        <div
          class="c7"
        >
          Is Featured
        </div>
        <div
          class="c8"
        >
          No
        </div>
      </div>
      <div
        class="c6"
      >
        <div
          class="c7"
        >
          Gallery Title
        </div>
        <div
          class="c8"
        >
          Ryerson Library Reading Room
        </div>
      </div>
      <div
        class="c6"
      >
        <div
          class="c7"
        >
          Type of Exhibition
        </div>
        <div
          class="c8"
        />
      </div>
    </div>
  </div>
</div>
`;
